flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCheckpointIDCounter.java

```diff
@@ -149,6 +150,11 @@ public void setCount(long newId) throws Exception {
 		sharedCount.setCount((int) newId);
 	}
 
+	@VisibleForTesting
+	ConnectionState getLastState() {
+		return connStateListener.lastState;
+	}
```

Comment:
I think this exposes internal details which are not relevant for the test. I would try to write the test without exposing these internals.